lib/ui/page/home/page/my_profile/delete_email/controller.dart

```diff
@@ -62,9 +60,18 @@ class DeleteEmailController extends GetxController {
         s.editable.value = false;
         s.status.value = RxStatus.loading();
         try {
-          await _myUserService.removeUserEmail(email, password: password);
-          pop?.call();
+          final code = ConfirmationCode.tryParse(s.text);
+          if (code == null) {
+            await _myUserService.removeUserEmail(email, password: password);
+          } else {
+            try {
+              await _myUserService.removeUserEmail(email, confirmation: code);
+            } catch (e) {
+              await _myUserService.removeUserEmail(email, password: password);
+            }
```

Comment:
`e` may be any exception. You shouldn't decide that `code` is not a `ConfirmationCode` based on __any__ exception.